python/refactor/nn/modules/module.py

```diff
@@ -0,0 +1,77 @@
+from refactor.nn.loss.loss_generator import LossGenerator
+from refactor.nn.metrics import MetricsEvaluator
+from collections import OrderedDict
+
+# encoder, decoder are both inherited from Module. The default behavior of get_loss(), evaluate()
+# does nothing as sometimes encoder does not need them. While for decoder, it's mandatory for it
+# to overload these two functions.
+
+
+class Module(object):
```

Comment:
are we trying to not rely on Pytorch?

Reply:
Resolved, I've converted it into nn.Module